src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Subscriptions.cs

```diff
@@ -0,0 +1,239 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Maestro.ScenarioTests.ObjectHelpers;
+using Microsoft.DotNet.Maestro.Client.Models;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace Maestro.ScenarioTests
+{
+    public class ScenarioTests_Subscriptions : MaestroScenarioTestBase
+    {
+        private TestParameters _parameters;
+
+        private string repo1Name = "maestro-test1";
+        private string repo2Name = "maestro-test2";
+        private string repo3Name = "marstro-test3";
+        private string channel1Name = "subscriptionTestChannel1";
+        private string channel2Name = "subscriptionTestChannel2";
+
+        [SetUp]
+        public async Task InitializeAsync()
+        {
+            _parameters = await TestParameters.GetAsync();
+            SetTestParameters(_parameters);
+        }
+
+        [TearDown]
+        public Task DisposeAsync()
+        {
+            _parameters.Dispose();
+            return Task.CompletedTask;
+        }
+
+        [Test]
+        public async Task Subscriptions_EndToEnd()
```

Comment:
Wow... this test is REALLY long, and doesn't assert much... It does a lot of actions, doesn't check that they did what it expects... only way at the end that somewhere, in this 200 lines of tests, something didn't do what it was supposed to. :-(

This isn't your fault... the PS tests do this too, but it makes it basically impossible to tell why the tests are failing. :-(

Reply:
I absolutely agree. I'd like to split it into individual tests, but I think that's going to get messy with even more things running in parallel (especially with the subscription tests, which use the same 3 repos for everything & will fail on duplicates) and I'd rather have some version of all the powershell tests checked in than spend a bunch more time refactoring.